src/viking.js

```diff
@@ -1,11 +1,105 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength){
+        this.health = health;
+        this.strength = strength;
+    }
+    attack(){
+        return (this.strength);
+    }
+    receiveDamage(damage){
+     this.health -= damage
+    }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+     constructor(name, health, strength){
+        super(health, strength)
+        this.name = name;
+     }
+     receiveDamage (damage){
+        this.health -= damage
+        if(this.health > 0){
+            return `${this.name} has received ${damage} points of damage`; 
+        } else
+            return `${this.name} has died in act of combat`;
+
+     }
+     battleCry(){
+        return `Odin Owns You All!`
+     }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier{
+    receiveDamage(damage){
+        this.health -= damage
+        if(this.health > 0){
+            return `A Saxon has received ${damage} points of damage`;
+        }else 
+            return `A Saxon has died in combat`;
+    }
+}
 
 // War
-class War {}
+class War {
+    constructor(){
+       this.vikingArmy = [];
+       this.saxonArmy = [];
+    }
+    addViking(Viking){
+        this.vikingArmy.push(Viking)
+
+    }
+
+    addSaxon(Saxon){       
+        this.saxonArmy.push(Saxon)
+    }
+    vikingAttack(){
```

Comment:
Here, the vikingAttack() function isn't working because you're not properly referencing the Viking army. Also, it uses Viking.strength to attack the Saxon, instead of getting a random Viking from the Viking army and using its attack method. And you're not calling splice correctly. It should be called on the saxonArmy array and passing in the index of the Saxon to remove and the number of items to remove (1) as arguments, not indexSaxon and indexSaxon.

This should fix it:

```
vikingAttack() {
        let indexViking = Math.floor(Math.random() * this.vikingArmy.length);
        let randomViking = this.vikingArmy[indexViking];
        let indexSaxon = Math.floor(Math.random() * this.saxonArmy.length);
        let randomSaxon = this.saxonArmy[indexSaxon];
        let saxonDamage = randomSaxon.receiveDamage(
          randomViking.attack()
        );
        if (randomSaxon.health <= 0) {
          this.saxonArmy.splice(indexSaxon, 1);
        }
        return saxonDamage;
      }
```